sources/languages/euphoria/description.md

```diff
@@ -38,11 +38,58 @@ is actually this:
 {65, 66, 67}
 ```
 
-Also according to [Wikipedia](https://en.wikipedia.org/wiki/Euphoria_(programming_language)#Parameter_passing)):
+[User-specified data types](https://openeuphoria.org/docs/lang_decl.html#_123_userdefinedtypes)
+can be done based on these four data types. The way this is done is to declare a type
+function that validates the input value. For example, if you wanted to define an
+`hour` type that only allowed integers from 0 to 23, you would do this:
+
+```euphoria
+type hour(integer x)
+    return x >= 0 and x <= 23
+end type
+```
+
+Keywords in Euphoria are rather simple. For each type of keyword that does control flow or
+declares a `function` or `procedure` (more on that later), there is a corresponding `end`
+keyword that declares the end of that structure. For example:
+
+```euphoria
+if x = 42
+then
+    do_something_cool(x)
+end if
+```
+
+Euphoria makes the distinction between functions and procedures. Functions returns
```

Comment:
I really like this feature. I had written about the differences between [statements and expressions in the past](https://therenegadecoder.com/code/the-difference-between-statements-and-expressions/), and this feature really drives home that distinction.